cardinal/world.go

```diff
@@ -0,0 +1,168 @@
+package cardinal
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/rs/zerolog/log"
+	"pkg.world.dev/world-engine/cardinal/ecs"
+	"pkg.world.dev/world-engine/cardinal/ecs/inmem"
+	"pkg.world.dev/world-engine/cardinal/ecs/storage"
+	"pkg.world.dev/world-engine/cardinal/ecs/transaction"
+	"pkg.world.dev/world-engine/cardinal/server"
+	"pkg.world.dev/world-engine/cardinal/shard"
+)
+
+type World struct {
+	impl *ecs.World
+}
+
+// System is a function that process the transaction in the given transaction queue.
+// Systems are automatically called during a world tick, and they must be registered
+// with a world using AddSystem or AddSystems.
+type System func(*World, *TransactionQueue) error
+
+type (
+	// EntityID represents a single entity in the World. An EntityID is tied to
+	// one or more components.
+	EntityID = storage.EntityID
+	TxHash   = transaction.TxHash
+)
+
+// NewWorldUsingRedis creates a new World object using Redis as the storage layer.
+func NewWorldUsingRedis(addr, password string, opts ...WorldOption) (*World, error) {
```

Comment:
See comments below, following that I think we can just call this `NewWorld`

Reply:
done